src/hal/sensor/sensor.cpp

```diff
@@ -0,0 +1,80 @@
+
+#include "../../hal/hal.h"
+#include "../../hal/sensor/sensor.h"
+
+// Sensor pin defines
+#define FLOW_SENSE_PIN     A0
+#define PRESSURE_SENSE_PIN A1
+
+
+// The following function initializes Timer 5 to do pressure and flow sensor data collection
+int16_t sensorHalInit(void)
+{
+  // Sensor timer
+  TCCR5A=0; // set timer control registers
+  TCCR5B=0x0A;  // set timer to clear timer on compare (CTC), and prescaler to divide by 8
+  TCNT5=0;  // initialize counter to 0
+  OCR5A= 1000;  // Set interrupt to 2khz
+  TIMSK5 |= 0x02; // enable timer compare interrupt
+
+  return HAL_OK;
+}
+
+
+//--------------------------------------------------------------------------------------------------
+int16_t pSum; // pressure data accumulator
+// This routine reads the pressure value as a voltage off the analog pin and inserts it into the filter
+void pressureSensorHalFetch(){
+  int16_t pIn;  // pressure value from sensor.
+  pIn=analogRead(PRESSURE_SENSE_PIN);
+  pSum = pSum-(pSum>>4)+pIn; // filter
+}
+
+// This routine returns the pressure value out of the filter
+int16_t pressureSensorHalGetValue(int16_t *value){
+  int32_t temp = pSum>>4;
+  int32_t pascals = ((temp * 217L) - 110995L)>>2;	// convert to Pascals
```

Comment:
Need to wrap the grabbing of pSum in an `ATOMIC_BLOCK` to avoid problems of interrupts coming in between the bytes since this is an 8bit processor after all.
```
  // Standard avr-libc include for atomic operations
  #include <util/atomic.h>
  ...
  ATOMIC_BLOCK(ATMOIC_RESTORESTATE) {
  int32_t temp = pSum>>4;
  }
  int32_t pascals = ((temp * 217L) - 110995L)>>2;
```

Reply:
Done